### PR TITLE
Term Entry - .mesh3d() - Plotly

### LINKED ISSUE #5178: [Term Entry] Python Plotly- graph_objects .Mesh3d()
### Reason/inspiration (optional)

We would like a new term entry on the `.Mesh3d()` method of the [`graph_objects module`](https://www.codecademy.com/resources/docs/plotly/graph-objects) in Plotly, Python. The entry should go in a new file under `docs/content/plotly/concepts/graph_objects/terms/mesh3d/mesh3d.md`

The entry should include the following:

- A description of the term
- A ##Syntax section that includes the syntax of the method and its details
- An ##Example section that includes a ```py ``` codeblock and ```shell ``` that shows the example output.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by danitellini**
### Description

Summary of Change:

Topic: Python Plotly `graph_object` `.mesh3d()` function
Files created: `docs/content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md`
Files added: `docs/media/mesh3d-example-result.png` and `docs/media/mesh3d-dataset-sphere.png`
Description: Created a new **Term Entry** for Python Plotly `graph_object` `.mesh3d()` function

Open issue, and a great way to practice and learn is to teach!

### Issue Solved

Closes #5178 

### Type of Change

- Adding a new entry
- ~Editing an existing entry (fixing a typo, bug, issues, etc)~
- ~Updating the documentation~

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by danitellini** [comment]
@mamtawardhani I have no idea my last commit became the title of the pull request 🤕 This one has beat my head in for sure haha

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  - 'Plotly'
+  - 'Python'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+  - 'paths/data-science'
+  - 'paths/data-science-foundations'
+---
+
+**`.Mesh3d()`** is a method in Plotly's `graph_objects` module that generates 3D mesh plots by defining vertices and connecting faces (triangles).
```
Comment (mamtawardhani) [inline content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:19]:
```suggestion
**`.Mesh3d()`** is a method in Plotly's [`graph_objects`](https://www.codecademy.com/resources/docs/plotly/graph-objects) module that generates 3D mesh plots by defining vertices and connecting them with faces (typically triangles).
```

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    x=x_values,
+    y=y_values,
+    z=x_values,
+    i=indices_i,    # Optional
+    j=indices_j,    # Optional
+    k=indices_k,    # Optional
+    color='color_value',    # Optional
+    opacity=opacity_value,  # Optional
+    (...)   # Optional
+)
```
Comment (mamtawardhani) [inline content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:34]:
```suggestion
plotly.graph_objects.Mesh3d( x=None, y=None, z=None, i=None, j=None, k=None, color=None, opacity=None, ...)   
```

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+```
+
+- `x`, `y`, `z`: Arrays of vertex coordinates
+- `i`, `j`, `k`: Arrays defining the triangles via indices pointing to the vertex arrays
+- `color`: Sets the mesh color
+- `opacity`: Sets the mesh transparency (0 to 1)
```
Comment (mamtawardhani) [inline content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:40]:
```suggestion
- `x`, `y`, `z`: Coordinates of the vertices in 3D space (arrays or lists).
- `i`, `j`, `k`: Indices that define the triangular faces, where each set of three indices corresponds to a triangle formed by the vertices defined by `x`, `y`, and `z`.
- `Color`: The color of the mesh, which can be a single color or an array of colors for each face.
- `opacity`: Controls the transparency of the mesh, ranging from 0 (fully transparent) to 1 (fully opaque).
```

Comment (mamtawardhani) [reply content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:40]: reframed and added some details

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md (hunk 2 above)
Comment (mamtawardhani) [reply content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:34]: default values are `None` which should be mentioned

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+)
+
+# Generate figure.
+fig = go.Figure(data=[mesh])
+
+# Display plot.
+fig.show()
+```
+
+> Note: If you don't specify the triangle indices, the mesh will automatically be generated using algorithms like Delaunay triangulation. You can also use the `alphahull` parameter to control the generation without the indices: `alphahull = -1` - Uses Delauney triangulation; `alphahull = 0` - Computes the convex hull of the point set; `alphahull > 0` - Uses the alpha shape algorithm with the specified alpha value.
```
Comment (mamtawardhani) [inline content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:82]:
```suggestion
> Note: If triangle indices (`i`, `j`, `k`) are not specified, the mesh will be automatically generated using algorithms like Delaunay triangulation. The `alphahull` parameter can be used to control the mesh generation process without the indices:
- `alphahull = -1`: Uses Delaunay triangulation.
- `alphahull = 0`: Computes the convex hull of the point set.
- `alphahull > 0`: Uses the alpha shape algorithm with the specified alpha value.
```

Comment (mamtawardhani) [reply content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:82]: We don't usually use you and we

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+This example results in the following output:
+
+![Example 2 Output]("https://raw.githubusercontent.com/Codecademy/docs/main/media/mesh3d-example-results.png")
+
+### Example 2
+
+Dataset information can also be used to create the mesh. This visualizes a sphere by using vertex and face data from a CSV file:
```
Comment (mamtawardhani) [inline content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:90]:
```suggestion
 The dataset information can be used to create a 3D mesh plot, as demonstrated in the example below, which visualizes a sphere using vertex and face data from a CSV file:
```

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+Tags:
+  - 'Graphics'
+  - 'Graphs'
+  - 'Plotly'
+  - 'Python'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+  - 'paths/data-science'
+  - 'paths/data-science-foundations'
```
Comment (mamtawardhani) [inline content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:16]:
```suggestion
```

Comment (mamtawardhani) [reply content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:16]: Let us only have 2 paths

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @danitellini thank you for contributing to Codecademy Docs, the entry is nicely written! 😄 

I've suggested a few changes, could you please review and modify those at your earliest convenience? Thank you! 😃

**Comment by mamtawardhani** [comment]
Hey @danitellini are you still working on this?

**Comment by danitellini** [comment]
Yes ma'am, I'll have the updates done later today. Work was heavy this week 😅

**Comment by mamtawardhani** [comment]
> Yes ma'am, I'll have the updates done later today. Work was heavy this week 😅

No problemo, was just checking 😃

**Comment by danitellini** [comment]
@mamtawardhani All set for next step!

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing to Codecademy Docs @danitellini 😄 

The entry looks good for a second round of review! 🚀

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+fig = go.Figure(data=[mesh])
+
+# Display plot.
+fig.show()
+```
+
+If triangle indices (`i`, `j`, `k`) are not specified, the mesh will be automatically generated using algorithms like Delaunay triangulation. The `alphahull` parameter can be used to control the mesh generation process without the indices:
+- `alphahull = -1`: Uses Delaunay triangulation.
+- `alphahull = 0`: Computes the convex hull of the point set.
+- `alphahull > 0`: Uses the alpha shape algorithm with the specified alpha value.
```
Comment (PragatiVerma18) [inline content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md]: The Delaunay triangulation explanation could be more informative here for a beginner.

Also, we should divide the text into smaller paragraphs to make it easy to follow.

```suggestion
If triangle indices (`i`, `j`, `k`) are not specified, the mesh will be automatically generated using algorithms like **Delaunay triangulation**.

_Delaunay triangulation_ is a method for dividing a set of points into non-overlapping triangles, ensuring that no points lie inside the circumcircle of any triangle. 

The `alphahull` parameter can be used to control the mesh generation process without the indices:
- `alphahull = -1`: Uses Delaunay triangulation.
- `alphahull = 0`: Computes the convex hull of the point set.
- `alphahull > 0`: Uses the alpha shape algorithm with the specified alpha value.
```

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+If triangle indices (`i`, `j`, `k`) are not specified, the mesh will be automatically generated using algorithms like Delaunay triangulation. The `alphahull` parameter can be used to control the mesh generation process without the indices:
+
+- `alphahull = -1`: Uses Delaunay triangulation.
+- `alphahull = 0`: Computes the convex hull of the point set.
+- `alphahull > 0`: Uses the alpha shape algorithm with the specified alpha value.
+
+This example results in the following output:
+
+![Example 2 Output]("https://raw.githubusercontent.com/Codecademy/docs/main/media/mesh3d-example-results.png")
```
Comment (PragatiVerma18) [inline content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:76]: The image description should be more descriptive.

```suggestion
![The output will be a 3D triangular pyramid (tetrahedron) visualized using Plotly.]("https://raw.githubusercontent.com/Codecademy/docs/main/media/mesh3d-example-results.png")
```

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+    k=df['k'],
+    facecolor=df['facecolor']
+))
+
+fig.show()
+```
+
+This example results in the following output:
+
+![Example 2 Output]("https://raw.githubusercontent.com/Codecademy/docs/main/media/mesh3d-dataset-sphere.png")
```
Comment (PragatiVerma18) [inline content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:105]: The image description should be more descriptive.

```suggestion
![The output will be a 3D sphere.]("https://raw.githubusercontent.com/Codecademy/docs/main/media/mesh3d-dataset-sphere.png")
```

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+- `alphahull = 0`: Computes the convex hull of the point set.
+- `alphahull > 0`: Uses the alpha shape algorithm with the specified alpha value.
+
+This example results in the following output:
+
+![Example 2 Output]("https://raw.githubusercontent.com/Codecademy/docs/main/media/mesh3d-example-results.png")
+
+## Example 2
+
+The dataset information can be used to create a 3D mesh plot, as demonstrated in the example below, which visualizes a sphere using vertex and face data from a CSV file:
```
Comment (PragatiVerma18) [inline content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:80]: We should mention the CSV file separately as well.

```suggestion
The dataset information can be used to create a 3D mesh plot, as demonstrated in the example below, which visualizes a sphere using vertex and face data from a [CSV file](https://raw.githubusercontent.com/plotly/datasets/master/ply/sphere-ply.csv):
```

**Review by PragatiVerma18 — CHANGES_REQUESTED** [review]
Hey @danitellini, the entry looks mostly good to me, I just have a few suggestions, please have a look and do the required changes.

**Comment by danitellini** [comment]
All set! @PragatiVerma18

**Review by PragatiVerma18 — APPROVED** [review]
This looks good to go! 🚀

**Pull request merged**